frontendUi/src/main/java/com/virtuslab/gitmachete/frontend/ui/cell/BranchOrCommitCellRenderer.java

```diff
@@ -136,26 +136,23 @@ void customize(BranchOrCommitCell cell, boolean isSelected, boolean hasFocus, in
       }
     }
 
-    /*
-     * TODO (#98): The padding is calculated for all commits in the same branch (and the branch itself). The time
-     * consumed could be reduced by using some lazy value, caching, or indent storing.
-     */
     @UIEffect
-    private int calculateTextPadding(IGraphElement element) {
-      int width = graphImage.getWidth();
-      if (element instanceof CommitElement) {
-        RepositoryGraph repositoryGraph = graphTable.getModel().getRepositoryGraph();
-        Collection<? extends PrintElement> printElements = repositoryGraph
-            .getPrintElements(((CommitElement) element).getBranchElementIndex());
-        double maxIndex = getMaxGraphElementIndex(printElements);
-        width = (int) (maxIndex * PaintParameters.getNodeWidth(graphTable.getRowHeight()));
-      }
+    private int calculateTextPadding(int maxPosition) {
+      int width = (maxPosition + 1) * PaintParameters.getNodeWidth(graphTable.getRowHeight());
       return width + LabelPainter.RIGHT_PADDING.get();
     }
 
+    private int getMaxGraphNodePositionInRow(IGraphNode node) {
+      // if node is a child (non root) branch the text must be shifted for sake of the right edge
+      // if node is a commit the text must be shifted for sake of its the shifted branch
+      boolean isRootBranch = node.isBranch() && ((BranchNode) node).getBranch().isRootBranch();
+      return node.getIndentLevel() + (!isRootBranch ? 1 : 0);
+    }
+
     @UIEffect
-    private GraphImage getGraphImage(Collection<? extends PrintElement> printElements) {
-      double maxIndex = getMaxGraphElementIndex(printElements);
+    private GraphImage getGraphImage(Collection<? extends PrintElement> printElements,
+        int maxGraphNodePositionInRow) {
+      double maxIndex = maxGraphNodePositionInRow;
```

Comment:
This local var is likely redundant (in fact in both contexts that you use it... is there a need for a `double`? maybe `long` instead?)